rxandroidble/src/main/java/com/polidea/rxandroidble/ClientComponent.java

```diff
@@ -119,28 +118,50 @@ static int provideDeviceSdk() {
         }
 
         @Provides
-        @Named(NamedSchedulers.GATT_CALLBACK)
+        @Named(NamedSchedulers.BLUETOOTH_INTERACTION)
         @ClientScope
-        static ExecutorService provideGattCallbackExecutorService() {
+        static ExecutorService provideBluetoothInteractionExecutorService() {
             return Executors.newSingleThreadExecutor();
         }
 
         @Provides
-        @Named(NamedSchedulers.GATT_CALLBACK)
+        @Named(NamedSchedulers.BLUETOOTH_CALLBACKS)
         @ClientScope
-        static Scheduler provideGattCallbackScheduler(@Named(NamedSchedulers.GATT_CALLBACK) ExecutorService executorService) {
-            return Schedulers.from(executorService);
+        static ExecutorService provideBluetoothCallbacksExecutorService() {
+            return Executors.newSingleThreadExecutor();
         }
 
         @Provides
-        LocationManager provideLocationManager() {
-            return (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+        @Named(NamedSchedulers.BLUETOOTH_INTERACTION)
+        @ClientScope
+        static Scheduler provideBluetoothInteractionScheduler(@Named(NamedSchedulers.BLUETOOTH_INTERACTION) ExecutorService service) {
+            return Schedulers.from(service);
+        }
+
+        @Provides
+        @Named(NamedSchedulers.BLUETOOTH_CALLBACKS)
+        @ClientScope
+        static Scheduler provideBluetoothCallbacksScheduler(@Named(NamedSchedulers.BLUETOOTH_CALLBACKS) ExecutorService service) {
+            return Schedulers.from(service);
         }
 
         @Provides
-        @Named(NamedSchedulers.MAIN_THREAD)
-        static Scheduler provideMainThreadScheduler() {
-            return AndroidSchedulers.mainThread();
+        static ClientComponentFinalizer provideFinalizationCloseable(
+                @Named(NamedSchedulers.BLUETOOTH_INTERACTION) final ExecutorService interactionExecutorService,
+                @Named(NamedSchedulers.BLUETOOTH_CALLBACKS) final ExecutorService callbacksExecutorService
+        ) {
+            return new ClientComponentFinalizer() {
+                @Override
+                public void onFinalize() {
+                    interactionExecutorService.shutdown();
```

Comment:
Actually this is incorrect. Dagger will provide you a new instance of the executor each time you ask for it. You seems to be closing an instance that wasn't even started and you're leaving started service as is. 

Provided executors should be scoped.